tests/e2e/utils/product-editor.js

```diff
@@ -577,17 +577,15 @@ export function getProductBlockEditorUtils( page ) {
 		async assertUnableSave( message = 'Please enter a valid value.' ) {
 			await this.clickSave();
 
-			const failureNotice = page
-				.locator( '.components-snackbar__content' )
-				.filter( { hasText: new RegExp( message ) } );
-
-			const failureNoticeDismissButton =
-				failureNotice.getByRole( 'button' );
+			const failureNotice = page.locator(
+				'.components-snackbar__content'
+			);
 
 			await expect( failureNotice ).toBeVisible();
+			await expect( failureNotice ).toHaveText( message );
 
-			// Dismiss the notice.
-			await failureNoticeDismissButton.click();
+			// // Dismiss the notice.
+			await failureNotice.click();
```

Comment:
This action failed with WC < 9.2. How about making it also compatible?

```suggestion
			// Dismiss the notice.
			await failureNotice
				.getByRole( 'button' ) // compatibility-code "WC < 9.2" -- Dismiss by its inner close button.
				.or( failureNotice )
				.click();

```

Reply:
I tried using the or method, but if both elements are present, it throws a "strict mode violation." Using nth to select one element worked for some tests but not for others. So, I decided to take a different approach, but you can adjust it if needed. [51e3679](https://github.com/woocommerce/google-listings-and-ads/pull/2511/commits/51e36797ff3f70569e16539409aed4990bd5120d)

Reply:
> I tried using the or method, but if both elements are present, it throws a "strict mode violation."

I later realized it had to be changed to this but I missed updating to the comment before submitting review.

```js
// Dismiss the notice.
await failureNotice
	.or( failureNotice.getByRole( 'button' ) )
	.last()
	.click();
```

> So, I decided to take a different approach, but you can adjust it if needed. [51e3679](https://github.com/woocommerce/google-listings-and-ads/pull/2511/commits/51e36797ff3f70569e16539409aed4990bd5120d)

Your method is better and clearer, thanks!